framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -33,7 +34,9 @@ std::shared_ptr<cCmduHeader> CmduMessageTx::create(uint16_t id, eMessageType mes
 std::shared_ptr<cCmduHeader> CmduMessageTx::load()
 {
     msg.reset(true);
-    return addClass<cCmduHeader>();
+    auto cmdu_header = msg.addClass<cCmduHeader>();
+    msg.addClass<ieee1905_1::tlvUnknown>();
```

Comment:
But what happens if it has more than 1 TLV? I don't think it will work.
I guess it will work for the specific CMDUs that we are using the certification buffer for, but maybe adding it in a loop until getting EOM might be a more correct solution. Of course, we will have to put 4 more zeros on the buffer, right after the CMDU we copied.

But generally, it is a GOOD solution and I like it. 😏

Reply:
That's the beauty of this solution - we don't care how many TLVs the CMDU contains.
Assume the buffer contents are a CMDU message with 3 TLVs: `[CMDU HEADER][TLV1][TLV2][TLV3]`
Then `load()` will treat `TLV1..TLV3` as one big list of unknown length without changing the buffer contents which is exactly what we want.
What it will do is to update `m_buff_ptr__` so on `finalize()` it will add the EOM TLV exactly at the end of the buffer, which is again, what we want.
And lastly, since it is an unknown length list of uint8_t, finalize will not swap the message contents which is again what we want since they are already swapped.
So bottom line, this covers all cases.

Reply:
After I looked on the init function of `tlvUnknown` it seems to me that it will only add one tlv.

Reply:
You are right, we need to add skipClass or something like that instead of unknown tlv.

Reply:
I removed this commit cause it was wrong - great catch, it worked by sheer luck!
I added 2 new commits which allow us to do this correctly and being able to verify the CMDU since getClass will now work.